public/app/features/variables/query/operators.ts

```diff
@@ -183,5 +181,26 @@ export function areMetricFindValues(data: any[]): data is MetricFindValue[] {
   }
 
   const firstValue: any = data[0];
-  return metricFindValueProps.some((prop) => firstValue.hasOwnProperty(prop) && typeof firstValue[prop] === 'string');
+  if (isDataFrame(firstValue)) {
+    return false;
+  }
+
+  for (const firstValueKey in firstValue) {
+    if (
+      !firstValue.hasOwnProperty(firstValueKey) ||
+      typeof firstValue[firstValueKey] === 'object' ||
+      typeof firstValue[firstValueKey] === 'function' ||
+      typeof firstValue[firstValueKey] === 'symbol'
```

Comment:
What is an allowed return type?

Reply:
not nested sub objects & functions. If it's an object that has a value prop that is string or number, and a text prop that is a string )

Reply:
Thank you, I've changed it to string or number. Regarding text and value, I've seen results with only the text property so I'm just checking if anyone of them exists.